trade-finance-manager-ui/server/controllers/case/amendments/bankDecision.controller.js

```diff
@@ -0,0 +1,323 @@
+    return res.redirect(`/case/${dealId}/underwriting`);
+  }
+};
+
+/**
+ * @param {*} req
+ * @param {*} res
+ * gets template for effective date for banks decision
+ */
+const getAmendmentBankDecisionEffectiveDate = async (req, res) => {
+  const { amendmentId, facilityId } = req.params;
+  const { data: amendment } = await api.getAmendmentById(facilityId, amendmentId);
+
+  if (!amendment?.amendmentId || !amendment?.bankDecision?.decision || !amendment?.bankDecision?.receivedDate) {
+    return res.redirect('/not-found');
+  }
+
+  const { user } = req.session;
+  const isEditable = userCanEditBankDecision(amendment, user) && amendment.status === AMENDMENT_STATUS.IN_PROGRESS;
+
+  // converts date from epoch format to day month year if in database
+  const effectiveDateDay = amendment?.bankDecision?.effectiveDate ? format(fromUnixTime(amendment.bankDecision.effectiveDate), 'dd') : '';
+  const effectiveDateMonth = amendment?.bankDecision?.effectiveDate ? format(fromUnixTime(amendment.bankDecision.effectiveDate), 'MM') : '';
+  const effectiveDateYear = amendment?.bankDecision?.effectiveDate ? format(fromUnixTime(amendment.bankDecision.effectiveDate), 'yyyy') : '';
+
+  return res.render('case/amendments/amendment-add-banks-decision-effective-date.njk', {
+    amendment,
+    isEditable,
+    user,
+    bankDecisionDateDay: effectiveDateDay,
+    bankDecisionDateMonth: effectiveDateMonth,
+    bankDecisionDateYear: effectiveDateYear,
+  });
+};
+
+/**
+ * @param {*} req
+ * @param {*} res
+ * posts effective date of bank decision and redirects to next page or renders template with errors
+ */
+const postAmendmentBankDecisionEffectiveDate = async (req, res) => {
+  const { _id: dealId, amendmentId, facilityId } = req.params;
+  const { body } = req;
+  const { user } = req.session;
+
+  const { data: amendment } = await api.getAmendmentById(facilityId, amendmentId);
+
+  if (!amendment?.amendmentId || !amendment?.bankDecision?.decision) {
+    return res.redirect('/not-found');
+  }
+
+  const isEditable = userCanEditBankDecision(amendment, user) && amendment.status === AMENDMENT_STATUS.IN_PROGRESS;
+
+  const type = 'bankDecisionDate';
+  const message = 'Enter the date the amendment will be effective from';
+
+  const { errorsObject, amendmentBankDecisionDateErrors, amendmentBankRequestDate } = await amendmentBankDecisionDateValidation(body, type, message);
+
+  if (amendmentBankDecisionDateErrors.length) {
+    return res.render('case/amendments/amendment-add-banks-decision-effective-date.njk', {
+      amendment,
+      isEditable,
+      user,
+      errors: errorsObject.errors,
+      bankDecisionDateDay: errorsObject.amendmentBankDecisionDateDay,
+      bankDecisionDateMonth: errorsObject.amendmentBankDecisionDateMonth,
+      bankDecisionDateYear: errorsObject.amendmentBankDecisionDateYear,
+    });
+  }
+
+  try {
+    const payload = { bankDecision: { effectiveDate: amendmentBankRequestDate } };
```

Comment:
should this be `amendmentBankEffectiveDate` ? and not `amendmentBankRequestDate`?

Reply:
updated, was using it as it came from the function (using function twice), but have renamed it